RELEASING.md

```diff
@@ -1,34 +1,74 @@
 # Releasing
 
-## Go SDK
+> **Warning**
+> Currently, if any SDK bumps an engine dependency, we should bump the engine dependency of all other SDKs concurrently. This helps ensure that users running multiple SDKs from a single machine all use the same engine. This is important as currently our automatic provisioning code enforces the existence of a single engine running at a time.
+
+## Engine
 
 ### Prep
 
+- Ensure you have GHCR credentials for your Dagger Github account configured ([instructions](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)).
+- Ensure [Mage is installed](https://magefile.org/).
+- Ensure that CI is green for all tests in main
+
 ```console
 git pull
 git status  # make sure everything is clean
 ```
 
-### Release Engine
+### Bump version tags
+
+- Modify `EngineImageRef` in `internal/mage/engine.go` to use a tag set to the engine version (e.g. `v0.3.0`).
 
 ```console
 export ENGINE_VERSION=v0.3.0
 git tag $ENGINE_VERSION
 git push origin $ENGINE_VERSION
+mage engine:publish
 ```
 
-### Bump Engine dependency
+## Go SDK
+
+### Prep
+
+```console
+git pull
+git status  # make sure everything is clean
+```
+
+### Bumping an Engine dependency
+
+The following is needed if the SDK requires a new version of the engine.
 
 ```console
-export ENGINE_VERSION=v0.3.0
 cd ./sdk/go
-go get github.com/dagger/dagger@${ENGINE_VERSION}
-go mod tidy
 git checkout -b go-sdk-update
-git commit go.mod go.sum -m 'sdk: go: bump engine dependency'
+```
+
+1. Obtain the full image ref, including SHA, from our GHCR repo for the version of the engine the SDK should depend on.
+
+- Our GHCR repo can be found [here](https://github.com/orgs/dagger/packages/container/package/engine).
+- Locate the tag for the engine the SDK should depend on and note the digest of it.
+  - The digest will be prefixed with `sha256:` and look like `a10730b1d305644b87c21630d27d9e15d424014ee7ac9bd44f4504fcfe92a8ca`
+
+1. Update the value of `engineImageRef` in `client.go` to be `ghcr.io/dagger/engine:<version tag>@sha256:<digest>` ([line currently here](https://github.com/dagger/dagger/blob/349f8a1661f1a8422a8bef919a345bd6dd3f3d85/sdk/go/client.go#L19))
```

Comment:
This is my favourite image ref format too: `<version>@sha256:<digest>`

I propose that we default to this whenever possible.